### PR TITLE
Auth and Basic Post functionalities

### PR DESCRIPTION
**Pull request by poornima-alugubelly**
Added 
Auth : user can login and signup 
Post : User can create post , edit post and delete post ( only text posts ) , like and dislike post

### CODE REVIEW

**Review by simran61 — APPROVED** [review]
Looks good
Nice folder structure

File: src/backend/controllers/AuthController.js [hunk 1]
```diff
@@ -15,47 +15,56 @@ const sign = require("jwt-encode");
...
-        {},
-        {
-          errors: ["Unprocessable Entity. Username Already Exists."],
-        }
-      );
-    }
-    const _id = uuid();
+	// console.log("inside sign up");
+	// console.log(JSON.parse(request.requestBody));
+	// check ...rest issue later
```
Comment (manojasarsa) [inline src/backend/controllers/AuthController.js:20]: You can remove these comments

File: src/pages/Auth/Login.jsx [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+	const { pwdToggle, pwdToggler } = usePwdToggler();
+	const [formVal, setFormVal] = useState({ username: "", password: "" });
+	const dispatch = useDispatch();
+	const { token, user, authStatus, authError } = useSelector(
+		(state) => state.auth
+	);
+	const navigate = useNavigate();
+
+	const loginHandler = async (e, username, password) => {
+		console.log(username, password);
```
Comment (manojasarsa) [inline src/pages/Auth/Login.jsx:33]: you can remove console.log, if no longer needed

File: src/reducers/postSlice.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+});
+
+export const editPost = createAsyncThunk(
+	"posts/editPost",
+	async ({ token, postId, postData }) => {
+		try {
+			const response = await editPostService(token, postId, postData);
+			return response.data.posts;
+		} catch (error) {
+			console.log(error.response.data);
```
Comment (manojasarsa) [inline src/reducers/postSlice.js:28]: you can use `console.error` to handle errors

**Review by manojasarsa — APPROVED** [review]
Rest the code is pretty organized and well structured with proper indentation.
You can remove the comments if no longer in use.
GTG!

**Pull request merged**